public/app/features/dashboard-scene/addToDashboard/AddToDashboardForm.tsx

```diff
@@ -33,36 +32,21 @@ interface SaveToExistingDashboard extends SaveTargetDTO {
 
 type FormDTO = SaveToNewDashboardDTO | SaveToExistingDashboard;
 
-function assertIsSaveToExistingDashboardError(
-  errors: FieldErrors<FormDTO>
-): asserts errors is DeepMap<SaveToExistingDashboard, FieldError> {
-  // the shape of the errors object is always compatible with the type above, but we need to
-  // explicitly assert its type so that TS can narrow down FormDTO to SaveToExistingDashboard
-  // when we use it in the form.
-}
-
-function getDashboardURL(dashboardUid?: string) {
-  return dashboardUid ? `d/${dashboardUid}` : 'dashboard/new';
-}
-
-enum GenericError {
-  UNKNOWN = 'unknown-error',
-  NAVIGATION = 'navigation-error',
-}
-
-interface SubmissionError {
-  error: AddToDashboardError | GenericError;
-  message: string;
-}
-
-interface Props {
+export interface Props<TOptions = undefined> {
   onClose: () => void;
-  exploreId: string;
+  buildPanel: (options: TOptions) => Panel;
+  timeRange?: TimeRange;
+  options: TOptions;
```

Comment:
the AddToDashboardForm is now a bit more generic, you pass it a function buildPanel and options (options just gets passed to buildPanel). This is to enable custom "add to dashboard" form options by users of this component. We do not have this use case right now but I can foresee that being useful.